easybuild/easyblocks/n/neuron.py

```diff
@@ -60,85 +62,133 @@ def extra_options():
-        python_root = get_software_root('Python')
-        if python_root:
-            self.with_python = True
-            self.cfg.update('configopts', "--with-nrnpython=%s/bin/python" % python_root)
+            # specify path to InterViews if it is available as a dependency
+            interviews_root = get_software_root('InterViews')
+            if interviews_root:
+                self.cfg.update('configopts', "-DIV_DIR=%s" % interviews_root)
+            else:
+                self.cfg.update('configopts', "-DNRN_ENABLE_INTERVIEWS=OFF")
 
-        # determine host CPU type
-        cmd = "./config.guess"
-        (out, ec) = run_cmd(cmd, simple=False)
+            # no longer used it seems
+            self.hostcpu = ''
 
-        self.hostcpu = out.split('\n')[0].split('-')[0]
-        self.log.debug("Determined host CPU type as %s" % self.hostcpu)
+            # optionally enable support for Python as alternative interpreter
+            python_root = get_software_root('Python')
+            if python_root:
+                self.with_python = True
+                self.cfg.update('configopts', "-DNRN_ENABLE_PYTHON=ON -DPYTHON_EXECUTABLE=%s/bin/python" % python_root)
+                self.cfg.update('configopts', "-DNRN_ENABLE_MODULE_INSTALL=ON "
+                                "-DNRN_MODULE_INSTALL_OPTIONS='--prefix=%s'" % self.installdir)
 
-        # determine Python lib dir
-        self.pylibdir = det_pylibdir()
+            # determine Python lib dir
+            self.pylibdir = det_pylibdir()
 
-        # complete configuration with configure_method of parent
-        super(EB_NEURON, self).configure_step()
+            # complete configuration with configure_method of parent
+            CMakeMake.configure_step(self)
 
     def install_step(self):
         """Custom install procedure for NEURON."""
 
         super(EB_NEURON, self).install_step()
 
-        if self.with_python:
-            pypath = os.path.join('src', 'nrnpython')
-            try:
-                pwd = os.getcwd()
-                os.chdir(pypath)
-            except OSError as err:
-                raise EasyBuildError("Failed to change to %s: %s", pypath, err)
+        # with the CMakeMake, the python module is installed automatically
+        if LooseVersion(self.version) < LooseVersion('7.8'):
+            if self.with_python:
+                pypath = os.path.join('src', 'nrnpython')
+                try:
+                    pwd = os.getcwd()
+                    os.chdir(pypath)
+                except OSError as err:
+                    raise EasyBuildError("Failed to change to %s: %s", pypath, err)
 
-            cmd = "python setup.py install --prefix=%s" % self.installdir
-            run_cmd(cmd, simple=True, log_all=True, log_ok=True)
+                cmd = "python setup.py install --prefix=%s" % self.installdir
+                run_cmd(cmd, simple=True, log_all=True, log_ok=True)
```

Comment:
with Autotools or CMake, explicitly doing this setup.py is not required since quite some time. I believe this can be removed. ([here](https://github.com/neuronsimulator/nrn/blob/80e01ecd6b42aa5ce3fd8db7bc5df8b15f2def2a/.travis_build.sh#L31) is simple version of travis build script)